src/quickgui/attributes/qgsquickattributeformmodel.h

```diff
@@ -59,6 +59,7 @@ class QUICK_EXPORT QgsQuickAttributeFormModel : public QSortFilterProxyModel
     {
       ElementType = Qt::UserRole + 1, //!< User role used to identify either "field" or "container" type of item
       Name, //!< Field Name
+      Alias, //!< Field Alias
```

Comment:
Each field in the attribute form has properties such as name, widget type or alias (etc.). If a user defines `Alias` for a field, having this role provides access to this property. A use case can be that alias is used instead of name property for a label while composing an attribute form in QML.

Reply:
Yes, but how will the QML get a value for this role if it's not handled anywhere in the model? Don't you always get an invalid QVariant?

Reply:
In the qgsquickfeatureform, there is a ListView containing fields which has its model (AttributeFormModel) and delegate item. Deligate item uses model roles to get value - it suppose to call` QVariant data(const QModelIndex &index, int role) const` function. In this case, its not necessary to override it, so the one of QStandardItem is called. Therefore it is not explicitly present in qgsquickattributeformmodel.

https://github.com/qgis/QGIS/blob/d21eb3fbf7db950a7900cda12cdf0236207ea529/src/quickgui/plugin/qgsquickfeatureform.qml#L256
Note: fieldItem is defined just few lines below of selected line.

Reply:
Ok, let's ask the other way round: all other roles have their values assigned [here](https://github.com/qgis/QGIS/blob/d21eb3fbf7db950a7900cda12cdf0236207ea529/src/quickgui/attributes/qgsquickattributeformmodelbase.cpp#L253-L266). Why does Alias not need to be present there?

Reply:
Btw, the alias is even already available with `Name` where set (others will return the field name) and also available as a property of the `Field` object.